MystenLabs/sui
# [sui-test-validator] update sui-test-validator options and dockerfile

## Description 

Give the address configuration to the developer allows for more flexibility in starting `sui-test-validator`.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change

### Release notes

#### sui-test-validator change

**sui-test-validator** cmd:

```bash
$ ./target/release/sui-test-validator --help
Start a Sui validator and fullnode for easy testing

Usage: sui-test-validator [OPTIONS]

Options:
  -c, --config-dir <CONFIG_DIR>                Config directory that will be used to store network config, node db, keystore sui genesis -f --with-faucet generates a genesis config that can be
                                               used to start this process. Example: sui-test-validator --config-dir ~/.sui/sui_config We can use any config dir that is generated by the sui genesis
      --fullnode-rpc-addr <FULLNODE_RPC_ADDR>  Address to start the Fullnode RPC server on [default: 127.0.0.1:9000]
      --faucet-addr <FAUCET_ADDR>              Address to start the Sui faucet on [default: 127.0.0.1:9123]
      --indexer-rpc-addr <INDEXER_RPC_ADDR>    Address to start the Indexer RPC server on [default: 127.0.0.1:9124]
      --pg-port <PG_PORT>                      Port for the Indexer Postgres DB 5432 is the default port for postgres on Mac [default: 5432]
      --pg-host <PG_HOST>                      Hostname for the Indexer Postgres DB [default: localhost]
      --epoch-duration-ms <EPOCH_DURATION_MS>  The duration for epochs (defaults to one minute) [default: 60000]
      --with-indexer                           if we should run indexer
      --use-indexer-experimental-methods       TODO(gegao): remove this after indexer migration is complete
  -h, --help                                   Print help
  -V, --version                                Print version
```

1. start at `127.0.0.1`
```bash
./target/release/sui-test-validator
```

2. start at `0.0.0.0`
```bash
./target/release/sui-test-validator --fullnode-rpc-addr "0.0.0.0:9000" --faucet-addr "0.0.0.0:9123"
```


#### Add sui-test-validator Dockerfile

Hopefully the mystenlabs team will maintain a docker image of sui-test-validator, which is really important for developers.

```bash
docker run -id -p 9000:9000 -p 9123:9123 vladilenaksana/sui-test-validator:0.0.2 sui-test-validator --fullnode-rpc-addr "0.0.0.0:9000" --faucet-addr "0.0.0.0:9123"
```
